### PR TITLE
Use pidof install ps to check process existence

### LINKED ISSUE #2251: keadm error to judge EdgeCore running
**What happened**:
root@ubuntu:/opt/opensource/kubeedge# _output/local/bin/keadm join -e cloud:20000
Host has mosquit+ already installed and running. Hence skipping the installation steps !!!
Error: EdgeCore is already running on this node, please run reset to clean up first
Usage:
  keadm join [flags]

Examples:

keadm join --cloudcore-ipport=<ip:port address> --edgenode-name=<unique string as edge identifier>

  - For this command --cloudcore-ipport flag is a required option
  - This command will download and install the default version of pre-requisites and KubeEdge

keadm join --cloudcore-ipport=10.20.30.40:10000 --edgenode-name=testing123 --kubeedge-version=1.2.1

Flags:
      --certPath string                  The certPath used by edgecore, the default value is /etc/kubeedge/certs (default "/etc/kubeedge/certs")
  -s, --certport string                  The port where to apply for the edge certificate
      --cgroupdriver string              CGroupDriver that uses to manipulate cgroups on the host (cgroupfs or systemd), the default value is cgroupfs
  -e, --cloudcore-ipport string          IP:Port address of KubeEdge CloudCore
  -i, --edgenode-name string             KubeEdge Node unique identification string, If flag not used then the command will generate a unique id on its own
  -h, --help                             help for join
      --interfacename string             KubeEdge Node interface name string, the default value is eth0
      --kubeedge-version string          Use this key to download and use the required KubeEdge version
  -p, --remote-runtime-endpoint string   KubeEdge Edge Node RemoteRuntimeEndpoint string, If flag not set, it will use unix:///var/run/dockershim.sock
  -r, --runtimetype string               Container runtime type
  -t, --token string                     Used for edge to apply for the certificate

F1015 10:51:24.305048   12107 keadm.go:27] EdgeCore is already running on this node, please run reset to clean up first
goroutine 1 [running]:
k8s.io/klog.stacks(0xc0000cb400, 0xc000380000, 0x78, 0xcc)
        /opt/opensource/kubeedge/vendor/k8s.io/klog/klog.go:883 +0xb8
k8s.io/klog.(*loggingT).output(0x1fd0320, 0xc000000003, 0xc0000fa000, 0x1f430c1, 0x8, 0x1b, 0x0)
        /opt/opensource/kubeedge/vendor/k8s.io/klog/klog.go:834 +0x330
k8s.io/klog.(*loggingT).printDepth(0x1fd0320, 0xc000000003, 0x1, 0xc000161f68, 0x1, 0x1)
        /opt/opensource/kubeedge/vendor/k8s.io/klog/klog.go:706 +0x129
k8s.io/klog.(*loggingT).print(...)
        /opt/opensource/kubeedge/vendor/k8s.io/klog/klog.go:697
k8s.io/klog.Fatal(...)
        /opt/opensource/kubeedge/vendor/k8s.io/klog/klog.go:1264
main.main()
        /opt/opensource/kubeedge/keadm/cmd/keadm/keadm.go:27 +0x8a
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
```sh
root@ubuntu:/opt/opensource/kubeedge# ps -ef |grep edgecore
root     12097 24758  0 10:51 pts/33   00:00:00 sudo wget -k --no-check-certificate https://raw.githubusercontent.com/kubeedge/kubeedge/release-1.4/build/tools/edgecore.service
root     12098 12097  0 10:51 pts/33   00:00:00 wget -k --no-check-certificate https://raw.githubusercontent.com/kubeedge/kubeedge/release-1.4/build/tools/edgecore.service
root     12210 20970  0 10:52 pts/29   00:00:00 grep --color=auto edgecore
```
After kill the wget, `keadm join` runs successfully. I guest the method judging the edgecore running is also using the `ps -ef |grep edgecore` method.
**Environment**:
<details><summary>Kubernets Version</summary>

```console
$ kubelet --version
Kubernetes v1.18.6
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:58:53Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:51:04Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

<details><summary>CloudCore Version</summary>

```console
$ cloudcore --version
KubeEdge v1.4.0
```

</details>

<details><summary>EdgeCore Version</summary>

```console
$ edgecore--version
KubeEdge v1.4.0
```

</details>

**CloudSide Environment**:
<details><summary>Hardware configuration</summary>

```console
$ lscpu
# paste output here
```

</details>

<details><summary>OS</summary>

```console
$ cat /etc/os-release
# paste output here
```

</details>

<details><summary>Kernel</summary>

```console
$ uname -a
# paste output here
```

</details>

<details><summary>Go version</summary>

```console
$ go version
# paste output here
```

</details>

<details><summary>Others</summary>
</details>

**EdgeSide Environment**:
<details><summary>Hardware configuration</summary>

```console
$ lscpu
# paste output here
```

</details>

<details><summary>OS</summary>

```console
$ cat /etc/os-release
# paste output here
```

</details>

<details><summary>Kernel</summary>

```console
$ uname -a
# paste output here
```

</details>

<details><summary>Go version</summary>

```console
$ go version
# paste output here
```

</details>

<details><summary>Others</summary>
</details>

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Xiang Dai <long0dai@foxmail.com>

**What type of PR is this?**
 /kind bug

**What this PR does / why we need it**:

`ps | grep` would get unexpected result, use `pidof` instead and use `pkill` to kill process.

`pidof` and `pkill` no need to be extra installed.

**Which issue(s) this PR fixes**:

Fixes #2251

### CODE REVIEW

**Comment by llhuii** [comment]
How about using a pid file instead of `pidof`?

**Comment by daixiang0** [comment]
We do not generate a pid file.

**Comment by llhuii** [comment]
how about `systemctl status edgecore`?

File: edge/test/integration/scripts/execute.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ function cleanup() {
   sudo pkill edgecore || true
   while true; do
     sleep 1
-    ps faux | grep -q [e]dgecore || break
+    pidof dgecore >/dev/null || break
```
Comment (llhuii) [inline edge/test/integration/scripts/execute.sh:23]: Miss the e of **e**dgecore?

**Comment by daixiang0** [comment]
In user case, it could be started with single binary.

File: edge/test/integration/scripts/execute.sh [hunk 2]
```diff
@@ -20,7 +20,7 @@ function cleanup() {
   sudo pkill edgecore || true
   while true; do
     sleep 1
-    ps faux | grep -q [e]dgecore || break
+    pidof edgecore >/dev/null && break
```
[the remark this answers is not in the document]
Comment (fisherxu) [reply edge/test/integration/scripts/execute.sh]: Agreed

[the remark this answers is not in the document]
Comment (daixiang0) [reply edge/test/integration/scripts/execute.sh]: Done.

**Pull request merged**